Addition of a connectivity check module

paribeiro10 opened the pull request:
Addition of a connectivity check module meant to handle connectivity changes, plus the respective sample screen to showcase the feature. This is usually a common class replicated in many projects, meaning it qualifies for ADAL's acceleration purpose.

It makes use of a Broadcast Receiver for the `android.net.conn.CONNECTIVITY_CHANGE` filter. Furthermore, it has a utility class to query the connectivity type (WIFI, MOBILE) and if there is a connection available (true or false). All actions are logged to log cat when debugging as well.

paribeiro10 committed:
Addition of a connectivity check module meant to handle connectivity changes.
Addition of a simple sample screen to showcase the feature.

extmkv commented:
Maybe we can merge this module with the network module to avoid one more module.

What do you think about that? @jzeferino @paribeiro10 @NunoSSilva @PedroOkawa @Andrer757

jzeferino commented:
@extmkv It make sense to be a separated module since we can use it witout retrofit/network module.Yes it will be another module, but IMO its not a biggie. Regarding the code, will review it tomorow. @paribeiro10. Another thing @paribeiro10 try to avoid the separated commits regarding checkstyle/pmd, etc. thanks for the contrib.

paribeiro10 commented:
@jzeferino I will be more careful next time, thanks.
@extmkv I agree with @jzeferino on the separate module option. My understanding was that the adal-network handled API calls and this just "observes" device connectivity.

jzeferino commented on adal-connectivity/src/main/java/com/massivedisaster/adal/connectivity/ConnectionChangeReceiver.java:
```diff
@@ -0,0 +1,68 @@
+ * Abstract class meant to offer connectivity change detection capabilities,
+ * using the Android system's connection change system filter.
+ */
+@SuppressWarnings("unused") public abstract class ConnectionChangeReceiver extends BroadcastReceiver {
```
@paribeiro10 the `supress` is really needed here? There is another way to solve this other than supress?

paribeiro10 replied:
In this case, the `suppress` is unneeded.
I have the habit of adding these 'suppress' when developing to remove the lint warnings from Android Studio (when you create the class it is not used as of yet).
I will remove it.

jzeferino commented on adal-connectivity/src/main/java/com/massivedisaster/adal/connectivity/NetworkConstants.java:
```diff
@@ -0,0 +1,50 @@
+/** Utility class meant to hold all constants for the Network sub module. */
+@SuppressWarnings({"unused", "WeakerAccess"}) public final class NetworkConstants {
+
+    /** ########## Misc. ########## */
```
@paribeiro10 We don't use this type of comment in ADAL instance members. Use only the message. Example here: https://github.com/massivedisaster/ADAL/blob/master/adal-application-state/src/main/java/com/massivedisaster/adal/applicationstate/ApplicationStateManager.java#L38

paribeiro10 replied:
Duly noted. I will also change this.

jzeferino commented on adal-connectivity/src/main/java/com/massivedisaster/adal/connectivity/NetworkConstants.java:
```diff
@@ -0,0 +1,50 @@
+    /** ########## Misc. ########## */
+    public static final String LOG_TAG = "CADL-Network";
+
+    /** ########## Exception Messages ########## */
```
Same as https://github.com/massivedisaster/ADAL/pull/125/files#r137398680

paribeiro10 replied:
Duly noted. I will also change this.

jzeferino commented on adal-connectivity/src/main/java/com/massivedisaster/adal/connectivity/NetworkConstants.java:
```diff
@@ -0,0 +1,50 @@
+    /** ########## Exception Messages ########## */
+    public static final String ASSERTION_ERROR = "Instantiating utility class.";
+
+    /** ########## Error Messages ########## */
```
Same as https://github.com/massivedisaster/ADAL/pull/125/files#r137398680

paribeiro10 replied:
Duly noted. I will also change this.

jzeferino commented on adal-connectivity/src/main/java/com/massivedisaster/adal/connectivity/NetworkConstants.java:
```diff
@@ -0,0 +1,50 @@
+    /** ########## Error Messages ########## */
+    public static final String INVALID_CONTEXT_INSTANCE = "::Invalid Context instance: ";
+
+    /** ########## Information Messages ########## */
```
Same as https://github.com/massivedisaster/ADAL/pull/125/files#r137398680

paribeiro10 replied:
Duly noted. I will also change this.

jzeferino commented on adal-connectivity/src/main/java/com/massivedisaster/adal/connectivity/NetworkConstants.java:
```diff
@@ -0,0 +1,50 @@
+
+    /** Instantiates a new NetworkConstants. Private to prevent instantiation. */
+    private NetworkConstants() {
+        throw new AssertionError(ASSERTION_ERROR); // Throw an exception if this *is* ever called
```
@paribeiro10 This comment could be removed, otherwise, place it above the throw.

paribeiro10 replied:
Duly noted. I will also change this.

jzeferino commented on adal-connectivity/src/main/java/com/massivedisaster/adal/connectivity/NetworkUtils.java:
```diff
@@ -0,0 +1,75 @@
+import android.net.NetworkInfo;
+
+/** Utility class for the Network sub module. */
+@SuppressWarnings({"unused", "WeakerAccess"}) public final class NetworkUtils {
```
Same as https://github.com/massivedisaster/ADAL/pull/125/files#r137397807

paribeiro10 replied:
Same as https://github.com/massivedisaster/ADAL/pull/125#discussion_r137652941

jzeferino commented on adal-connectivity/src/main/java/com/massivedisaster/adal/connectivity/NetworkUtils.java:
```diff
@@ -0,0 +1,75 @@
+
+    /** Instantiates a new NetworkUtils. Private to prevent instantiation. */
+    private NetworkUtils() {
+        // Throw an exception if this *is* ever called
```
@paribeiro10 as opposed to https://github.com/massivedisaster/ADAL/pull/125/files#r137399198 the comment is present and above the throw.

paribeiro10 replied:
I removed it as it is not needed. Well spotted on that one.

jzeferino commented on adal-connectivity/src/main/java/com/massivedisaster/adal/connectivity/ConnectionChangeReceiver.java:
```diff
@@ -0,0 +1,68 @@
+    private Context mContext;
+
+    /**
+     * Procedure meant to handle this {@link BroadcastReceiver}'s registration to any {@link Activity} instance.
```
@paribeiro10 The way the method documentation start ins't the way ADAL has other methods comments.
Here is an example: https://github.com/massivedisaster/ADAL/blob/master/adal-alarm/src/main/java/com/massivedisaster/adal/alarm/AlarmManager.java#L81

Every method should have a short description, describing waht it does, like: "Cancel alarm added before".

Please take a look at all method documentation started by "Procedure".

paribeiro10 replied:
Duly noted. I will also change this.

jzeferino commented:
@paribeiro10 It could be done in this PR or in a new one, but please don't forget to add the docs about the sample.

Take a look at the [docs](https://github.com/massivedisaster/ADAL/tree/master/docs to create) for existing modules. 

Don't forget to later add the doc in the readme [Usage](https://github.com/massivedisaster/ADAL#usage) section

jzeferino commented on adal-connectivity/src/main/java/com/massivedisaster/adal/connectivity/NetworkConstants.java:
```diff
@@ -0,0 +1,63 @@
+/**
+ * Utility class meant to hold all constants for the Network sub module.
+ */
+@SuppressWarnings({"unused", "WeakerAccess"}) public final class NetworkConstants {
```
@paribeiro10 forgot this one?

paribeiro10 replied:
Duly noted. Removed.

jzeferino commented on docs/adal-connectivity.md:
```diff
@@ -0,0 +1,57 @@
+# adal-connectivity
+Android library to verify if the device has an Internet connection and what it's type is!
```
@paribeiro10 replacing "and what it's type is!" with "and what type it is" seems more clear to read.

paribeiro10 replied:
Duly noted. Replaced.

jzeferino commented on docs/adal-connectivity.md:
````diff
@@ -0,0 +1,57 @@
+  compile 'com.massivedisaster.adal:adal-connectivity:0.1.15'
+}
+```
+### Usage
````
@paribeiro10 the Usage should have activity/fragment/application/service code for the user to understand how to call this utility methods. I suggest a more simple usage such as calling the `checkConnectivity` and show its result in a `Toast`, for example.

paribeiro10 replied:
Made changes to the class to make it more clear for this purpose.
I can remove the `TextView` and leave the Toast only. Please advise.

jzeferino replied:
Your call.

jzeferino commented on sample/src/main/java/com/massivedisaster/adal/sample/feature/connectivity/FragmentConnectivityAware.java:
```diff
@@ -0,0 +1,127 @@
+     *
+     * @param isOnline boolean value indicating whether the device has a connection established or not.
+     */
+    private void handleConnectivityStatusChange(final boolean isOnline) {
```
@paribeiro10 This one its just a enhancement. You could remove the "duplicated" lines here, adding a ternary operator to get the string: 

```
String deviceConnectivity = isOnline ? getString(R.string.connectivity_device_online) : getString(R.string.connectivity_device_offline);

mTxtMessage.setText(Connectivity);
Log.d(getActivity().getClass().getName(), Connectivity);
```

Plus, the `Log` here seems irrelevant.

paribeiro10 replied:
Good catch. Replaced.

jzeferino commented on sample/src/main/res/layout/fragment_connectivity_aware.xml:
```diff
@@ -0,0 +1,42 @@
+        android:layout_centerInParent="true"
+        android:textColor="@color/colorPrimary"
+        android:textSize="22sp"
+        android:visibility="visible"/>
```
@paribeiro10 `android:visibility="visible"` Irrelevant.

paribeiro10 replied:
Removed.

jzeferino commented:
@paribeiro10 why did you changed the original travis configuration?

paribeiro10 committed:
Correction of the issues reported by the reviewer.

paribeiro10 commented:
@jzeferino I had a brain freeze last Tuesday. I removed the filler commits and corrected the issues.
The Travis configuration should be fine now. I changed it because I can't run the check plugin on my Windows machine... It got changed because of merge mix-ups.
